### ISSUE TITLE
read_html can't parse the output of to_html() when index has name

### ISSUE THREAD

**Issue by ruoyu0088** [pandas-dev/pandas#8897]
Here is the code:

```
import pandas as pd
import numpy as np
import io
df = pd.DataFrame(np.random.rand(6, 3), columns=list("ABC"))
df.index = list("abcdef")
df.index.name = "index"
df2 = pd.read_html(df.to_html(), index_col=0)
print df2[0]
```

Here is the output:

```
          A         B         C  index  Unnamed: 5  Unnamed: 6  Unnamed: 7
a  0.759803  0.307306  0.348905    NaN         NaN         NaN         NaN
b  0.003259  0.400688  0.859137    NaN         NaN         NaN         NaN
c  0.824059  0.817198  0.308370    NaN         NaN         NaN         NaN
d  0.067031  0.517689  0.354966    NaN         NaN         NaN         NaN
e  0.043291  0.388853  0.342978    NaN         NaN         NaN         NaN
f  0.064113  0.707188  0.971005    NaN         NaN         NaN         NaN
```

**Comment by clham** [comment]
Currently ``.to_html`` renders as follows:
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>A</th>
      <th>B</th>
      <th>C</th>
    </tr>
    <tr>
      <th>index</th>
      <th></th>
      <th></th>
      <th></th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>a</th>
      <td>0.042651</td>
      <td>0.664604</td>
      <td>0.673674</td>
    </tr>
    <tr>
      <th>b</th>
      <td>0.368379</td>
      <td>0.282512</td>
      <td>0.248967</td>
    </tr>
    <tr>
      <th>c</th>
      <td>0.044463</td>
      <td>0.258529</td>
      <td>0.462941</td>
    </tr>
    <tr>
      <th>d</th>
      <td>0.213760</td>
      <td>0.730625</td>
      <td>0.917187</td>
    </tr>
    <tr>
      <th>e</th>
      <td>0.588074</td>
      <td>0.158623</td>
      <td>0.251396</td>
    </tr>
    <tr>
      <th>f</th>
      <td>0.749541</td>
      <td>0.139867</td>
      <td>0.886418</td>
    </tr>
  </tbody>
</table>

Is that what is intended, or should it look like this:
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th>index</th>
      <th>A</th>
      <th>B</th>
      <th>C</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>a</th>
      <td>0.499722</td>
      <td>0.037782</td>
      <td>0.851000</td>
    </tr>
    <tr>
      <th>b</th>
      <td>0.880219</td>
      <td>0.619207</td>
      <td>0.914825</td>
    </tr>
    <tr>
      <th>c</th>
      <td>0.154390</td>
      <td>0.693243</td>
      <td>0.719830</td>
    </tr>
    <tr>
      <th>d</th>
      <td>0.508555</td>
      <td>0.259720</td>
      <td>0.865177</td>
    </tr>
    <tr>
      <th>e</th>
      <td>0.949483</td>
      <td>0.329493</td>
      <td>0.241324</td>
    </tr>
    <tr>
      <th>f</th>
      <td>0.038173</td>
      <td>0.768175</td>
      <td>0.838935</td>
    </tr>
  </tbody>
</table>

Note: The second version reads in properly, and looks more "table-like", however it is a different representation that what comes out when ipython displays a Dataframe.

Bottom Line:  Do we want change ``.to_html`` to write the bottom version, or change ``.read_html`` to accept the top version? I'm in favor of making the bottom version the standard. I'm happy to try to PR once it's decided.

**Comment by cpcloud** [comment]
@clham I think the rendering is a separate issue from the bug posted by @ruoyu0088. If you'd like to discuss different HTML rendering, please open another issue. Thanks. Marking as a bug.

**Comment by NumberPiOso** [comment]
As @clham stated we need to decide. Currently, df.to_html() produces
<table border="1" class="dataframe"> <thead> <tr style="text-align: right;">
        <tr> <th></th> <th>A</th> <th>B</th> <th>C</th> </tr>
        <tr> <th>index</th> <th></th> <th></th> <th></th> </tr>
    </thead> <tbody> <tr> <th>a</th> <td>0.137095</td> <td>0.929869</td> <td>0.283542</td> </tr> <tr>
            <th>b</th> <td>0.769681</td> <td>0.415683</td> <td>0.721674</td> </tr> </tbody></table>

But it we change the default behaviour to

<table border="1" class="dataframe"> <thead> <tr style="text-align: right;">
        <tr> <th>index</th> <th>A</th> <th>B</th> <th>C</th> </tr>
    </thead> <tbody> <tr> <th>a</th> <td>0.137095</td> <td>0.929869</td> <td>0.283542</td> </tr> <tr>
            <th>b</th> <td>0.769681</td> <td>0.415683</td> <td>0.721674</td> </tr> </tbody></table>

We can solve this BUG.
